src/Interpreters/Aggregator.h

```diff
@@ -1071,25 +1072,9 @@ class Aggregator final
     /// For external aggregation.
     void writeToTemporaryFile(AggregatedDataVariants & data_variants, size_t max_temp_file_size = 0) const;
 
-    TemporaryFileOnDiskHolder createTempFile(size_t max_temp_file_size) const;
+    bool hasTemporaryData() const { return tmp_data && !tmp_data->empty(); }
 
-    bool hasTemporaryFiles() const { return !temporary_files.empty(); }
-
-    struct TemporaryFiles
-    {
-        std::vector<TemporaryFileOnDiskHolder> files;
-        size_t sum_size_uncompressed = 0;
-        size_t sum_size_compressed = 0;
-        mutable std::mutex mutex;
-
-        bool empty() const
-        {
-            std::lock_guard lock(mutex);
-            return files.empty();
-        }
-    };
-
-    const TemporaryFiles & getTemporaryFiles() const { return temporary_files; }
+    TemporaryDataOnDisk & getTemporaryData() const { return *tmp_data; }
```

Comment:
As I found by searching through the diff, this method used only in AggregatingTransform, where the result can be made `const`, then may be return const reference?